arithmetization/src/main/java/net/consensys/linea/bl/TransactionProfitabilityCalculator.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package net.consensys.linea.bl;
+
+import java.math.BigDecimal;
+
+import lombok.extern.slf4j.Slf4j;
+import net.consensys.linea.config.LineaTransactionSelectorConfiguration;
+import org.hyperledger.besu.datatypes.Transaction;
+import org.hyperledger.besu.datatypes.Wei;
+import org.slf4j.spi.LoggingEventBuilder;
+
+@Slf4j
+public class TransactionProfitabilityCalculator {
+
+  private final LineaTransactionSelectorConfiguration conf;
+  private final double preComputedValue;
+
+  public TransactionProfitabilityCalculator(final LineaTransactionSelectorConfiguration conf) {
+    this.conf = conf;
+    this.preComputedValue =
+        conf.estimateGasMinMargin() * conf.gasPriceRatio() * conf.verificationGasCost();
+  }
+
+  public Wei profitablePriorityFeePerGas(
+      final Transaction transaction, final Wei minGasPrice, final long gas) {
+    final double txSize = Math.max(0, transaction.getSize() + conf.adjustTxSize());
```

Comment:
is there a scenario where `transaction.getSize()` or `conf.adjustTxSize()` can be negative?

Anyway it does not hurt.

Reply:
conf.adjustTxSize has historically be a negative constant -45, if the config change to another value, it's safer to use the Max here